sdk/keyvault/keyvault-keys/review/keyvault-keys.api.md

```diff
@@ -104,11 +108,11 @@ export class CryptographyClient {
     encrypt(algorithm: EncryptionAlgorithm, plaintext: Uint8Array, options?: EncryptOptions): Promise<EncryptResult>;
     get keyID(): string | undefined;
     sign(algorithm: SignatureAlgorithm, digest: Uint8Array, options?: SignOptions): Promise<SignResult>;
-    signData(algorithm: SignatureAlgorithm, data: Uint8Array, options?: SignOptions): Promise<SignResult>;
+    signData(algorithm: SignatureAlgorithm, data: Uint8Array, options?: SignDataOptions): Promise<SignDataResult>;
```

Comment:
This could work, but I was imaging doing something like we did with `encrypt` and `decrypt` above e.g., `signDataParameters: SignDataParameters` and making the local hashing function a required argument. What you're doing here is suitable as well, though. @tg-msft @bterlson what do you think? It's not really consistent with what we did above with `encrypt` and `decrypt`, but those who suggested we do that later didn't like it, so I'm confused what we should do in this case.

Reply:
I was playing around with that as well but I don't think it's strictly necessary here - there aren't many different permutations of parameters keyed by the algorithm like we do with AES encrypt / decrypt. In this case, the hashing algorithm is not required, it's an optional parameter with a sane default. Users who want to choose a different hashing algorithm have a few options:

1. Use `sign` and `verify` and create the digest yourself -or-
2. Use the new option we provided here

Happy to discuss alternatives, I just think that SignParameters might be unnecessary here

Reply:
> I just think that SignParameters might be unnecessary here

You're probably right, but I'm curious whether we should stick with the pattern we started, good or bad. I'm also worried about this point in time. Right now, seems SHA256 is all that's supported. But that's right now. What happens when it changes? If you look at a lot of crypto APIs, there's not a lot of default behaviors. The ramifications of those defaults or some heuristic behaviors can be disastrous.

I also replied on our internal Teams thread and would like to see what @herveyw-msft thinks as well.

Reply:
I spoke with @KrzysztofCwalina and he's fine with this pattern, though it got me to thinking we should probably populate the new result property with the hash algorithm we used for any `SignatureAlgorithm` e.g., SHA256 for ES256.